src/python/pants/backend/javascript/lint/prettier/rules.py

```diff
@@ -28,9 +28,9 @@
 
 @dataclass(frozen=True)
 class PrettierFmtFieldSet(FieldSet):
-    required_fields = (JSSourceField,)
+    required_fields = (JSRuntimeSourceField,)
```

Comment:
1. No. I have no use case myself but I suppose a contributor could open up prettier to more languages later by adding a union-layer? Alternatively prettier seems like the kind of tool one might want to run "target-less"? 
2. Yes, the typescript compiler can check other files than .ts. It can be ran on .js, .jsx and  .tsx as well. Test runners have the same capability, so the test goal also applies.

Reply:
I also think it is fun that you called json non-js shaped :smirk:

Reply:
> Alternatively prettier seems like the kind of tool one might want to run "target-less" and just glob?

Hm, maybe, yeah.

> The typescript compiler can check other files than .ts. It can be run on .js, .jsx and .tsx as well Test runners have the same capability, so the test goal also applies. Those are the usecases I'm primarily intrested in, and I think the "JS-ish source" model works well there.

👍

Reply:
I think I wrote the Prettier plugin a while back, and we did have the discussion about how it could be used target-less, but that (d)evolved into a much larger discussion about the needs for targets and target-less and so on and so forth.

There are a few tools where I think we just want to defer to whatever the tool can handle - and I don't know how we do that in the most pants-esque way